sui_programmability/examples/nft_standard/sources/safe.move

```diff
@@ -0,0 +1,108 @@
+module nft_collection::safe {
+
+    /// A shared object for storing NFT's of type `T`, owned by the holder of a unique `OwnerCap`.
+    /// Permissions to allow others to list NFT's can be granted via TransferCap's and BorrowCap's
+    struct Safe<phantom T> has key {
```

Comment:
An idea how to support custom royalty policies and make them enforcable via this object:

1. When a buyer pays for the NFT, the trading contract such as OB transfers the `Coin<FT>` they should receive to be owned to an object similar to `RoyaltyReceipt has store`.
2. The `RoyaltyReceipt` is stored in the `Safe` object.
3. The user must call some function in `T` reference by `Safe<T>` which knows how to execute the royalty policy. Only then can they access the `Coin<FT>`.

To recapitulate: the trading algorithms know nothing of royalties, they just make the `Coin<FT>` object child of some `RoyaltyToBeCollected`/`RoyaltyReceipt` object. The user can access the `Coin<FT>` via `T::resolve_royalty` fn which unwraps the `Coin<FT>`, takes royalty and gives the rest to the NFT seller.

Reply:
We can wrap the marketplace/wallet cut into this object too to incentivize them to perform the `T::resolve_royalty` function by making that function permission-less. The marketplace/wallet could either: _(i)_ include the call via a batched tx, _(ii)_ have automation that does this as a scheduled job.

That way we take away the responsibility for sending that tx from the user.